### ISSUE TITLE
AC display light switch not working

### ISSUE THREAD

**Issue by gabi1324** [ollo69/ha-smartthinq-sensors#475]
**Describe the bug**
Hello! I have a problem with the display light on an S09ET AC unit. The switch is always off and if I toggle it the unit beeps but doesn't do anything. I can control the light from the remote but not from the LG app.

**Expected behavior**
control the display light from HA

**Screenshots**

**Environment details:**
 - Environment (HASSIO, Raspbian, etc):
 - Home Assistant version installed:
 - Component version installed:
 - Last known working version:
 - LG device type and model with issue: AC (S09ET)
 - LG devices connected (list):

**Output of HA logs**
Paste the relevant output of the HA log here.

**Additional context**

**Comment by ollo69** [comment]
Please attach diagnostic

**Comment by gabi1324** [comment]
[config_entry-smartthinq_sensors-06a774f58ee56ffb645ea7e4f785c74f.json.txt](https://github.com/ollo69/ha-smartthinq-sensors/files/10380790/config_entry-smartthinq_sensors-06a774f58ee56ffb645ea7e4f785c74f.json.txt)

**Comment by SN0WTR00PER** [comment]
Same thing for me...  Its not working.
Here my diag file
[config_entry-smartthinq_sensors-fc6f45dbc8d08d7c44080f6b4f49c7f5.json.txt](https://github.com/ollo69/ha-smartthinq-sensors/files/10430696/config_entry-smartthinq_sensors-fc6f45dbc8d08d7c44080f6b4f49c7f5.json.txt)

**Comment by gabi1324** [comment]
Hi! Are there any updates on this issue?

**Comment by gabi1324** [comment]
Still an issue.

**Comment by vincentwolsink** [comment]
I also have issues with S09ET and S12ET (RAC_056905_WW-RAC) airco's. The display light button is present for both ACs. When toggling the button the ACs do make a beep sound, but nothing happens.

See also #515

**Comment by vifrost** [comment]
1. download statistic information ( device informaion )

![image](https://github.com/ollo69/ha-smartthinq-sensors/assets/67081596/32da31b1-e9f6-441a-baa5-181a480e80ec)

2. find "airState.lightingState.displayControl"
3. check number of @AC_LED_OFF_W and @AC_LED_ON_W
4. edit ac.py
( /root/config/custom_components/smartting_sensors/wideq/devices/ac.py )
5. working right ON/OFF 

![image](https://github.com/ollo69/ha-smartthinq-sensors/assets/67081596/e95b429a-2f80-4b96-b174-a7851f841370)

**Comment by gabi1324** [comment]
Hi! I tried modifying ac.py but it doesn't work. The AC unit beeps and doesn't do anything. In the diagnostic file, I have _"airState.lightingState.displayControl": 0.0_ . I think this modification works for other AC models, but it isn't working for S09ET and other different variations.

**Comment by SiboutVanLoo** [comment]
Hi there, tried it to, but does'nt help, but seems to point in the right direction.
Changed the setting to 11 and 12, like you suggested, and now the 'light-state' is displayed correctly, but the command does'nt seem to get thru.
If i turn it off by remote controle, it shows "off" after a few minutes, 5 i guess, like the code suggests.
So that's a good thing, so any ideas why the "command" aspect seems to stay behind?
Was looking in sections further ahead in the code, mentioning "display", but my python knowledge seems to be lacking, and i would'nt want to wreck anything.
Thanks!

**Comment by SiboutVanLoo** [comment]
Ok,
This is the return information of the AC unit.

"airState.lightingState.displayControl": 1.0,
            "airState.wDir.hStep": 3.0,
            "mid": 6982764.0,
            "airState.energy.onCurrent": 60.0,
            "airState.wMode.airClean": 1.0,
            "airState.quality.sensorMon": 0.0,
            "airState.tempState.target": 25.0,

Above is Display light off (From aircon data.)

"airState.lightingState.displayControl": 0.0,
              "airState.wDir.hStep": 3.0,
              "mid": 7716930.0,
              "airState.energy.onCurrent": 152.0,
              "airState.wMode.airClean": 1.0,
              "airState.quality.sensorMon": 0.0,
              "airState.tempState.target": 24.0,
              
This was display ON


Below is "airclean data "on" from unit

"airState.lightingState.displayControl": 0.0,
              "airState.wDir.hStep": 3.0,
              "mid": 179596276.0,
              "airState.energy.onCurrent": 146.0,
              "airState.wMode.airClean": 0.0,
              "airState.quality.sensorMon": 0.0,
              "airState.tempState.target": 23.0,

Below is airclean data "off" from unit.

"airState.lightingState.displayControl": 0.0,
              "airState.wDir.hStep": 3.0,
              "mid": 179391791.0,
              "airState.energy.onCurrent": 146.0,
              "airState.wMode.airClean": 1.0,
              "airState.quality.sensorMon": 0.0,
              "airState.tempState.target": 23.0,

Here is the airclean mapping data from unit.

              "airState.wMode.airClean": {
                "data_type": "enum",
                "default": "0",
                "value_mapping": {
                  "0": "@AC_MAIN_AIRCLEAN_OFF_W",
                  "1": "@AC_MAIN_AIRCLEAN_ON_W"

Below is the mapping data of the display.

"airState.lightingState.displayControl": {
                "data_type": "enum",
                "default": "0",
                "value_mapping": {
                  "0": "@RAC_LED_ON",
                  "1": "@RAC_LED_OFF",
                  "2": "@NON",
                  "3": "@NON",
                  "4": "@NON",
                  "5": "@NON",
                  "6": "@NON",
                  "7": "@NON",
                  "8": "@NON",
                  "9": "@NON",
                  "10": "@NON",
                  "11": "@AC_LED_OFF_W",
                  "12": "@AC_LED_ON_W",
                  "13": "@AC_LED_AUTO_W"

Here is the python command setting code.

ADD_FEAT_POLL_INTERVAL = 300  # 5 minutes

LIGHTING_DISPLAY_OFF = "0"
LIGHTING_DISPLAY_ON = "1"

MODE_OFF = "@OFF"
MODE_ON = "@ON"

MODE_AIRCLEAN_OFF = "@AC_MAIN_AIRCLEAN_OFF_W"
MODE_AIRCLEAN_ON = "@AC_MAIN_AIRCLEAN_ON_W"


"airclean" works fine, but display does not.

Changing 1 & 0 to 
LIGHTING_DISPLAY_OFF = "@RAC_LED_OFF"
LIGHTING_DISPLAY_ON = "@RAC_LED_ON"
Or
LIGHTING_DISPLAY_OFF = "@AC_LED_OFF_W"
LIGHTING_DISPLAY_ON = "@AC_LED_ON_W"
Or
LIGHTING_DISPLAY_OFF = "1.0"
LIGHTING_DISPLAY_ON = "0.0"
Or
LIGHTING_DISPLAY_OFF = 1
LIGHTING_DISPLAY_ON = 0
Or
LIGHTING_DISPLAY_OFF = 1.0
LIGHTING_DISPLAY_ON = 0.0

Did not work.
Restarted HA every change, reloaded the thinq integration.

**Comment by filipef101** [comment]
The homebridge lge plugin works as expected, could be worth taking a look at what they are doing

**Comment by SiboutVanLoo** [comment]
Bump

**Comment by ollo69** [comment]
The problem with controls is that I need the right command to use.
Someone should use a mitm connetion to capture this. There is a good guide to so this, as reported [here](https://github.com/ollo69/ha-smartthinq-sensors#obtaining-api-information). If someone can do this, that I can try to fix this issue.

**Comment by ollo69** [comment]
> There isn't an option for 'display light' in the app, there is a button on the physical remote control though which works.

But I suppose that remote use infrared, doesn't it? So this should not be possible to implement at all. 
The only way that is that someone with an AC device play with the integration code [here](https://github.com/ollo69/ha-smartthinq-sensors/blob/4195bc5fa0f7eb0b6cb4f4787c69e1092f40f9b9/custom_components/smartthinq_sensors/wideq/devices/ac.py#L795), trying to find a command that control the light. In case I will be happy to implement in a new release.

**Comment by elpigeondindo** [comment]
> > There isn't an option for 'display light' in the app, there is a button on the physical remote control though which works.
> 
> But I suppose that remote use infrared, doesn't it? So this should not be possible to implement at all. The only way that is that someone with an AC device play with the integration code [here](https://github.com/ollo69/ha-smartthinq-sensors/blob/4195bc5fa0f7eb0b6cb4f4787c69e1092f40f9b9/custom_components/smartthinq_sensors/wideq/devices/ac.py#L795), trying to find a command that control the light. In case I will be happy to implement in a new release.

I can try but I don't know how to code in python too much. If someone tells me what to do I can try

**Comment by Coubi028** [comment]
> @ollo69 There isn't an option for 'display light' in the app, there is a button on the physical remote control though which works.

Yes I tried too and there is no option to activate/deactivate the light in the LG Thinq androïd App...

**Comment by dougiteixeira** [comment]
Hey @ollo69, I saw that [this feature was removed in the v0.35.0 version](https://github.com/ollo69/ha-smartthinq-sensors/commit/ff535cc01fd868edbad5ed14452e5023b1cd7f94) due to several open issues reporting that it doesn't work... but here it worked perfectly.

I want to see with you whether it is acceptable to implement this as a service so that users who have compatible equipment can continue to use the feature to turn the display light on and off. I understand that making it available as a service will reduce complaints that it doesn't work due to the non-existence of the entity, but they can still happen, so in the documentation I would make it clear that it is not compatible with all available models.

What do you think?

**Comment by ollo69** [comment]
@dougiteixeira,

you are the first reporting that the switch is working. Can you please share your integration diagnostic so that I can try to find a way to identify devices that support this feature?

Thanks👍

**Comment by dougiteixeira** [comment]
With the conditioner turned off, the display remains off even with the switch on, but when the conditioner is turned on, the display follows the switch configuration.

Integration Diagnostics: [smartthinq_sensors-cc226181134f606f36942198ac83891b-Ar Quarto-731c136139db78a958b86733d019cdd3.json (5).txt](https://github.com/ollo69/ha-smartthinq-sensors/files/13236300/smartthinq_sensors-cc226181134f606f36942198ac83891b-Ar.Quarto-731c136139db78a958b86733d019cdd3.json.5.txt)

Debug logs during commands: [home-assistant-smartthinq-sensors.log](https://github.com/ollo69/ha-smartthinq-sensors/files/13236294/home-assistant-smartthinq-sensors.log)

**Comment by ldotlopez** [comment]
Hi @ollo69 ,

I came here from the notice in the update README:
"Remove light display switch entity for AC (generally not working, see #475)"

That switch is working for me too, and I'm using it in some automations.

Here are my diagnostics, feel free to ask anything.
[smartthinq_sensors-8163e2cb1a99ab2500664e17e12b061f-Aire acondicionado habitación-1b16701a9d97af16e21eb981de606784.json.txt](https://github.com/ollo69/ha-smartthinq-sensors/files/13236313/smartthinq_sensors-8163e2cb1a99ab2500664e17e12b061f-Aire.acondicionado.habitacion-1b16701a9d97af16e21eb981de606784.json.txt)

**Comment by ollo69** [comment]
The main reason to disable the switch was to find you guys😊.
I will take a look to this on next days and I'll give you a feed-back.

**Comment by rodolfomachado** [comment]
Hi guys,
+1 to say that The switch was working fine.
Btw, I use it a lot. Bit it only works after the ac in ON. If the AC is off will just make a sound.
If you need any diagnostic I can send you

**Comment by pedropaislopes** [comment]
Hi! +1 here! Display switch working fine for 7 AC units on the 0.34.0 version. But for 2 of them the display logic is inverted - off on HA display on, on unit. I can open a new issue related to this inverted behavior once display switch is back!

**Comment by ollo69** [comment]
> If you need any diagnostic I can send you

Yes please, more diagnostics I have for working device better I can identify working devices. Till now I understood that switch work only when AC is on.

**Comment by ollo69** [comment]
> But for 2 of them the display logic is inverted

Please provide diagnostics for both devices type, 1 with correct logic and 1 with inverted logic. Thanks.

**Comment by rodolfomachado** [comment]
[error_log.txt](https://github.com/ollo69/ha-smartthinq-sensors/files/13237486/error_log.txt)
As someone said, it does not work when it's off but if you turn it on it will follow the switch

**Comment by selectcarleasing** [comment]
Just to add, on mine when the unit is on this switch will make the unit make a noise but nothing happens with the light.

**Comment by mukaschultze** [comment]
Just to add my two cents here, the switch works for me when the unit is both on and off, although the switch is inverted (on turns the light off, and vice versa). Model number is S4NW09AA31B.EB2GAMZ. I'll see if I can get the diagnostic data later.

**Comment by rodolfomachado** [comment]
Previously I sent debug logging, now The diagnostics. For both ACs The switch were working as I said 
[smartthinq_sensors-b841fba9bca1284b3335c14ff0eed2bc-Ar da Suíte-2f5ba8c6fc513baf97b91265cd24c5bd.json](https://github.com/ollo69/ha-smartthinq-sensors/files/13242530/smartthinq_sensors-b841fba9bca1284b3335c14ff0eed2bc-Ar.da.Suite-2f5ba8c6fc513baf97b91265cd24c5bd.json)
[smartthinq_sensors-b841fba9bca1284b3335c14ff0eed2bc-Ar da Lara-bc88583922fc5b9b571aa319a6845856.json](https://github.com/ollo69/ha-smartthinq-sensors/files/13242532/smartthinq_sensors-b841fba9bca1284b3335c14ff0eed2bc-Ar.da.Lara-bc88583922fc5b9b571aa319a6845856.json)

**Comment by ldotlopez** [comment]
> Just to add, on mine when the unit is on this switch will make the unit make a noise but nothing happens with the light.



> But it only works after the ac in ON. If the AC is off will just make a sound.

Same for my unit.

**Comment by ollo69** [comment]
@All,

I just restored the switch with last release, adding some more controls in the way that the current status is detected. For now I do not consider the AC status (on/off) so that anybody can test switch behavior. In any case looking at the attached diagnostic files the switch had to be reversed for all in previous versions, I would be surprised if it didn't.
Please report here results with your devices using last release, more details you will provide better will be.

**Comment by dougiteixeira** [comment]
@ollo69, I tested version 0.35.1 here, but the swicth is inverted in relation to the real status of the display.

Diagnostic files:
* [Switch ON - Display OFF.txt](https://github.com/ollo69/ha-smartthinq-sensors/files/13244549/Switch.ON.-.Display.OFF.txt)
* [Switch OFF - Display ON.txt](https://github.com/ollo69/ha-smartthinq-sensors/files/13244550/Switch.OFF.-.Display.ON.txt)

**Comment by dougiteixeira** [comment]
> For now I do not consider the AC status (on/off) so that anybody can test switch behavior.

I believe that the switch should remain active regardless of the conditioner being turned on, as the configuration will be changed in the same way, but the physical display will only be turned on if the conditioner is in operation and the switch is active.

In other words, I can change the setting regardless of the mode the conditioner is in (on/off).

And when we use the remote control, this behavior occurs as well.

**Comment by SiboutVanLoo** [comment]
I see a "Bell sound" section...
Interesting! 😃
But off topic, thanks for looking in to it @ollo!

Op do 2 nov 2023 22:15 schreef dougiteixeira ***@***.***>:

**Comment by ollo69** [comment]
> but the swicth is inverted in relation to the real status of the display.

I'm not understanding how to determinate this. Let's wait for others feedback.

**Comment by rodolfomachado** [comment]
Well, I'm not at home till Sunday to test but I can say that was usefull to have the switch even when the AC was off. Another point, to me the switch was not inverted, so no need to invert the logic

**Comment by ollo69** [comment]
It should be great have diagnostic from someone that say that logic is inverted, as @mukaschultze or @pedropaislopes

**Comment by mukaschultze** [comment]
@ollo69 here you go, diagnostic data and logs when toggling the switch

[ac-off-light-off.json](https://github.com/ollo69/ha-smartthinq-sensors/files/13245030/ac-off-light-off.json)
[ac-off-light-on.json](https://github.com/ollo69/ha-smartthinq-sensors/files/13245031/ac-off-light-on.json)
[ac-on-light-off.json](https://github.com/ollo69/ha-smartthinq-sensors/files/13245032/ac-on-light-off.json)
[ac-on-light-on.json](https://github.com/ollo69/ha-smartthinq-sensors/files/13245034/ac-on-light-on.json)
[toggling-light.log](https://github.com/ollo69/ha-smartthinq-sensors/files/13245035/toggling-light.log)

**Comment by pedropaislopes** [comment]
Hi! Unfortunately I’m not at home until Saturday. When back home I can provide you with all logs!

**Comment by ollo69** [comment]
@All,

please test last release and let me know if switch logic is correct for all.

**Comment by ldotlopez** [comment]
> please test last release and let me know if switch logic is correct for all.

Logic is inverted for me now.
Also: if the unit is off light switch is on, toggling does nothing but a beep.

**Comment by dougiteixeira** [comment]
> Also: if the unit is off light switch is on, toggling does nothing but a beep.

This is because the setting is being changed successfully, but the display on some conditioners will only be limited if the conditioner is operating.

**Comment by dougiteixeira** [comment]
> Logic is inverted for me now. Also: if the unit is off light switch is on[1], toggling does nothing but a beep.
> 
> Edit: 1. The switch in on with the unit off at startup, after that it respects the (inverted) state.
> 
> [diagnostics-light-on-sensor-off.json.txt](https://github.com/ollo69/ha-smartthinq-sensors/files/13249043/diagnostics-light-on-sensor-off.json.txt) [diagnostics-light-off-sensor-on.json.txt](https://github.com/ollo69/ha-smartthinq-sensors/files/13249042/diagnostics-light-off-sensor-on.json.txt)

Hey @ldotlopez, version 0.35.2 was released which corrected this inversion for me, your diagnostic files show version 0.35.1, can you confirm if updating to version 0.35.2 makes any changes?

**Comment by ldotlopez** [comment]
> Hey @ldotlopez, version 0.35.2 was released which corrected this inversion for me, your diagnostic files show version 0.35.1, can you confirm if updating to version 0.35.2 makes any changes?

Thanks @dougiteixeira, I missed that second update.

It works OK now

**Comment by elpigeondindo** [comment]
For me no change. Unable to adjust the display. There is a beep but no change.

**Comment by elpigeondindo** [comment]
@ollo69 
Okay but how i can do this?

**Comment by ollo69** [comment]
Go on integration page and select download diagnostic.

**Comment by ollo69** [comment]
@elpigeondindo,

I checked your diagnostic but it is identical to the others, so I have no idea why is not working for you.

**Comment by ollo69** [comment]
I close this issue, everything should be done was done. If there are other issue or request please open a new thread.

**Comment by joaolucasd** [comment]
Just adding this here for people looking for a way arround with the model RAC_056905_WW. **I could not find the fix, but just a diagnosis.**
Although API is not working, when you press the light button on the IR controller, the state of the light changes when looking at the snapshot:
'airState.lightingState.displayControl': 0.0 = lights on
'airState.lightingState.displayControl': 1.0 = lights off
But when you send the command (Setting new state for device XXXXX: basicCtrl - Set - airState.lightingState.displayControl - 1 or Setting new state for device XXXXX: basicCtrl - Set - airState.lightingState.displayControl - 0) nothing happens, just the beep sound.
I guess the AC can show the light state, but cannot change it through API. That´s a big bummer, cause this AC is on my toddler's room and having that hability would've been awesome.

**Comment by filipef101** [comment]
Kind of random, but kind of related, has anyone tried to disabled the beeping sound? Some remotes have that option but mine doesn't assume AC doesn't support disabling the beeping 😔

**Comment by ollo69** [comment]
@All,

this was already discussed a lot. The light control is there (exactly the same used by other integration) but not all AC device support this control via API.
There is nothing that I can do, if the integration control just make a beep and nothing change means that the control doesn't work and the only thing you can do is disable it.

**Comment by luiscamalmeida** [comment]
Since I don't believe we'll ever get these functions added by LG let me share how I've modded the units around my house to solve those problems. For the beeping sound just take the cap of the piezo beeper and remove the metal disc (save it for a later date should you ever feel nostalgic). For he display, I've taped some aluminium tape on top of the display and it dims the light enough for it to be used on our son's bedroom.
